main.tf

```diff
@@ -19,11 +19,9 @@ data "aws_ssm_parameter" "this" {
 resource "aws_instance" "this" {
   count = local.create && !var.ignore_ami_changes && !var.create_spot_instance ? 1 : 0
 
-  ami                  = try(coalesce(var.ami, nonsensitive(data.aws_ssm_parameter.this[0].value)), null)
-  instance_type        = var.instance_type
-  cpu_core_count       = var.cpu_core_count
```

Comment:
removing these is a breaking change - we should be able to keep them since their defaults are `null` and just add the new `cpu_options` argument

Reply:
Well spotted thanks !
I hope this new commit addresses this as you suggested.